gaugan/models/blocks.py

```diff
@@ -1,80 +1,67 @@
 import tensorflow as tf
-import tensorflow_addons as tfa
+from tensorflow.keras import Sequential
 from tensorflow.keras import layers, initializers
+import tensorflow_addons as tfa
 
-from .spade import SpatialAdaptiveNormalization
+from .spade import SPADE
 
 
 class ResidualBlock(layers.Layer):
-    """Residual Block
-
-    Reference: https://arxiv.org/abs/1903.07291
-
-    Args:
-        n_filters (int): Number of filters in the conv layers
-    """
-
-    def __init__(self, n_filters: int, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.n_filters = n_filters
+    def __init__(self, filters, **kwargs):
+        super().__init__(**kwargs)
+        self.filters = filters
 
     def build(self, input_shape):
         input_filter = input_shape[-1]
-        self.spade_1 = SpatialAdaptiveNormalization(input_filter)
-        self.spade_2 = SpatialAdaptiveNormalization(self.n_filters)
-        self.conv_1 = layers.Conv2D(self.n_filters, 3, padding="same")
-        self.conv_2 = layers.Conv2D(self.n_filters, 3, padding="same")
+        self.spade_1 = SPADE(input_filter)
+        self.spade_2 = SPADE(self.filters)
+        self.conv_1 = layers.Conv2D(self.filters, 3, padding="same")
+        self.conv_2 = layers.Conv2D(self.filters, 3, padding="same")
         self.learned_skip = False
-        if self.n_filters != input_filter:
+
+        if self.filters != input_filter:
             self.learned_skip = True
-            self.spade_3 = SpatialAdaptiveNormalization(input_filter)
-            self.conv_3 = layers.Conv2D(self.n_filters, 3, padding="same")
+            self.spade_3 = SPADE(input_filter)
+            self.conv_3 = layers.Conv2D(self.filters, 3, padding="same")
 
     def call(self, input_tensor, mask):
         x = self.spade_1(input_tensor, mask)
         x = self.conv_1(tf.nn.leaky_relu(x, 0.2))
         x = self.spade_2(x, mask)
         x = self.conv_2(tf.nn.leaky_relu(x, 0.2))
```

Comment:
The Leaky ReLU alpha should come from a function argument otherwise the purpose of using a configuration-based setup would get demeaned.